---
title: "DAOY-SPHER_Seurat_EXPLORE1_50K"
author: "JJ"
date: "2023-01-26"
output: html_document
---

```{r graphical_output}
options(bitmapType='cairo-png') # this line of code needed BEFORE knitr code in chunk below
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi = 300,
                      echo = TRUE,   # prints the code (if FALSE then only code output is shown)
                      cache = TRUE,
                      fig.path = "./outputs50K/DAOY_SPHER/figures/")
```

```{r libraries}
suppressPackageStartupMessages({
library(Seurat)
library(glmGamPoi)
library(presto)
library(msigdbr)
library(fgsea)
library(org.Hs.eg.db)
library(DOSE)
library(clustree)
library(tidyverse)
library(magrittr)
library(RColorBrewer)
library(patchwork)
})
```

Type of analysis: clustering of the DAOY-SPHEROID sample, marker identification, cell cycle difference regression, proliferation scoring of clusters, pathway enrichment for clusters. Workflow performed EXcluding mitochondrial genes, with mitochondrial gene regression and INcluding mitochondrial genes - essentially the same results. At an early step any cells with >10% mitochondrial reads were excluded from further analyses.

```{r eval=FALSE, echo=FALSE}
# load the data from scratch
# load the DAOY-SPHER raw data (from "working_data" folder)
data_dir <- "./working_data/DAOY-SPHER/"
list.files(data_dir) # should show matrix.mtx.gz, features.tsv.gz, barcodes.tsv.gz
```

```{r eval=FALSE, echo=FALSE}
# note these are .gz files which can be directly read. n = 761 cells (in dySpher.data)
dySpher.data <- ReadMtx(mtx = paste0(data_dir, "matrix.mtx.gz"),
                         features = paste0(data_dir, "features.tsv.gz"), 
                         cells = paste0(data_dir, "barcodes.tsv.gz"))
# create the seurat object, keep genes expressed in 10 or more cells
dySpher <- CreateSeuratObject(counts = dySpher.data,
                          min.cells = 10, min.genes=200,
                          project = "dySpher")

dim(dySpher) # 16,238 genes and 761 cells
```

```{r eval=FALSE, echo=FALSE}
# identify mitochondrial genes
mito.genes <- grep(pattern = "^MT", x = rownames(x = dySpher), value = TRUE)
percent.mito <- 
  Matrix::colSums(dySpher[mito.genes, ])/Matrix::colSums(dySpher)

# add the mitochondrial gene percent to the metadata
dySpher <- 
  AddMetaData(object = dySpher, metadata = percent.mito, col.name = "percent.mito")

# remove all cells with percent.mito > 10%
dySpher <- subset(dySpher, subset = percent.mito < 0.1)
dim(dySpher) # reduced to 16,238 genes and 649 cells

# remove mitochondrial genes from the remaining cells in the dataset
dySpher <- dySpher[ ! grepl("^MT-", rownames(dySpher)), ]
dim(dySpher) # 16,225 genes and 649 cells
```

```{r eval=FALSE, echo=FALSE}
# save the NON-normalised, NON-clustered seurat object
saveRDS(dySpher, file = "./outputs50K/DAOY_SPHER/filt.MitoCellsGenes.RiboRetained.seurat.DAOYspher.rds")
```

```{r eval=FALSE, echo=FALSE}
dySpher <- readRDS("outputs50K/DAOY_SPHER/filt.MitoCellsGenes.RiboRetained.seurat.DAOYspher.rds")
```

```{r eval=FALSE, echo=FALSE}
# load cell cycle genes, calculate cell-cycle difference in advance of regressing out CC difference

# normalise object before running cell cycle scores as per Seurat github issue: https://github.com/satijalab/seurat/issues/1679
dySpher <- NormalizeData(dySpher)

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
# calculate CC difference
dySpher <- CellCycleScoring(dySpher,
                g2m.features = g2m.genes,
                s.features = s.genes)
dySpher$CC.difference <- dySpher$S.Score - dySpher$G2M.Score
```

```{r eval=FALSE, echo=FALSE}
# this object (dySpher.input.integration) is input for integration - already SCTransformed, cell cycle difference regressed, and RunPCA()
dySpher.input.integration <- SCTransform(dySpher, 
                   vars.to.regress = "CC.difference",
                   method = "glmGamPoi",
                   vst.flavor = "v2",
                   verbose = FALSE) %>% 
  RunPCA(npcs = 30, verbose = FALSE)
```

```{r eval=FALSE, echo=FALSE}
# saved the UNCLUSTERED SCTransformed object, CC difference regressed, PCA reduction, as input for integrated workflow
saveRDS(dySpher.input.integration, "./outputs50K/DAOY_SPHER/DAOYspher.inputIntegration.RiboRetained.SCTransformed.CCdiffRegressed.PCA.unclustered.rds")
```

```{r eval=FALSE, echo=FALSE}
# cluster the cells with sctransform workflow
dySpher <- SCTransform(dySpher, 
                       vars.to.regress = c("CC.difference"),
                       method = "glmGamPoi", verbose = FALSE) %>% 
  RunPCA(verbose = FALSE) %>% 
  RunUMAP(dims=1:30, verbose=FALSE) %>% 
  FindNeighbors(dims=1:30, verbose=FALSE) %>% 
  FindClusters(verbose=FALSE, resolution = 0.5)
```

```{r eval=FALSE, echo=FALSE}
# save the clustered, SCTransformed object above
saveRDS(dySpher, file = "./outputs50K/DAOY_SPHER/DAOYspher.MitoCellsGenesFilt.RiboRetained.SCT.clustered.CCdiffRegressed.rds")
```

```{r eval=FALSE, echo=FALSE}
# load the above object
dySpher <- readRDS("outputs50K/DAOY_SPHER/DAOYspher.MitoCellsGenesFilt.RiboRetained.SCT.clustered.CCdiffRegressed.rds")
```

```{r eval=FALSE, echo=FALSE, DimplotDAOYspherUmapSCTransform, fig.height=6, fig.width=10}
DimPlot(dySpher, label = TRUE, pt.size = 1, label.size = 4)
```

SECTION END
##########################################################################################################

##########################################################################################################
***Proliferation scoring of clusters***
```{r eval=FALSE, echo=FALSE}
# load the list of proliferation genes in Seurat
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
prolif.genes <- list(c(s.genes, g2m.genes))
```


```{r eval=FALSE, echo=FALSE, AddModuleScoreProlifernDAOYspheroid, fig.height=6, fig.width=10}
# Dimplot of UMAP clusters with proliferation score superimposed (as heatmap)
phase.dySpher <- AddModuleScore(dySpher,
                        features = prolif.genes,
                        name = "prolif.genes")
FeaturePlot(phase.dySpher, features = 'prolif.genes1', label = TRUE, repel = TRUE, pt.size = 1,
            label.size = 4) +
   scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "")
```

SECTION END
##########################################################################################################


##########################################################################################################
***Get the cluster markers***
```{r eval=FALSE, echo=FALSE}
# get the markers of each cluster
dySpher.POSmarkers <- FindAllMarkers(object = dySpher, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r eval=FALSE, echo=FALSE}
# extract only the top 30 markers of each cluster
# cluster 2 only is enriched for ribosome genes.
dySpher.POSmarkers %>% 
  dplyr::group_by(cluster) %>% 
  slice_head(n = 30)
```

```{r eval=FALSE, echo=FALSE}
# Check expression of granule markers: MEIS1 is expressed of these seven genes
dySpher.POSmarkers %>%
  filter(gene %in% c("NEUROD1", "CNTN1", "CNTN2", "MEIS1", "MEIS2", "DCX", "SATB2"))
```

```{r eval=FALSE, echo=FALSE, ViolinplotDAOYspheroidMKI67andMEIS1, fig.height=4, fig.width=8}
# rough-and-ready plot
VlnPlot(dySpher, features = c("MKI67", "MEIS1"))
```

```{r eval=FALSE, echo=FALSE, ViolinplotDAOYspheroidZicGenes, fig.height=4, fig.width=8}
VlnPlot(dySpher, features = c("ZIC2", "CNTN1", "CNTN2"))
```


```{r eval=FALSE, echo=FALSE, VlnPlotMoreGranuleGenesDAOYspher, fig.height=17, fig.width=22}
# formatted for figure publication
p1a <- VlnPlot(dySpher,
               features = "MEIS1") +
  theme(axis.text = element_text(size = 22),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        plot.title = element_text(size = 30))

p1b <- VlnPlot(dySpher,
               features = "CNTN2") +
  theme(axis.text = element_text(size = 22),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        plot.title = element_text(size = 30))

p1c <- VlnPlot(dySpher,
               features = "RTN1") +
  theme(axis.text = element_text(size = 22),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        plot.title = element_text(size = 30))

p1d <- VlnPlot(dySpher,
               features = "MEIS2") +
  theme(axis.text = element_text(size = 22),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        plot.title = element_text(size = 30))

p1e <- VlnPlot(dySpher,
               features = "ZIC2") +
  theme(axis.text = element_text(size = 22),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        plot.title = element_text(size = 30))

p1f <- VlnPlot(dySpher,
              features = "JKAMP") +
  theme(axis.text = element_text(size = 22),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        plot.title = element_text(size = 30))

p1a + p1b + p1c + p1d + p1e + p1f + plot_layout(ncol = 2)
```

```{r eval=FALSE, echo=FALSE}
# save the positive markers of DAOY_SPHER
saveRDS(dySpher.POSmarkers, 
        file = "./outputs50K/DAOY_SPHER/POS.markers.DAOYspheroid.RiboRetained.SCT.clustered.rds")
```

```{r eval=FALSE, echo=FALSE}
# load the positive markers
dySpher.POSmarkers <- readRDS("outputs50K/DAOY_SPHER/POS.markers.DAOYspheroid.RiboRetained.SCT.clustered.rds")
```

```{r eval=FALSE, echo=FALSE}
table(Idents(dySpher))
```

As with ONS76-spheroid cluster 4, the DAOY-spheroid cluster 4 expresses proliferation markers
```{r eval=FALSE, echo=FALSE, HeatmapDaoySpherTop10markers4clustersRes0.5, fig.height=8, fig.width=10}
# first load dyMono seurat object above, then only run this chunk
dySpher.POSmarkers %>% 
  group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(dySpher, features = top10$gene) + guides(color = "none") +
  theme(text = element_text(size = 25), legend.title = element_text(size=20), 
        legend.text=element_text(size=20))
```

SECTION END
##########################################################################################################


##########################################################################################################
***FGSEA to find gene ontology pathway enrichment for the clusters***
```{r eval=FALSE, echo=FALSE}
# prepare the data for gProfiler (web app) input - gProfiler output then used as Cytoscape input
dySpher.genes.presto <- wilcoxauc(dySpher, 'seurat_clusters')
nrow(dySpher.genes.presto)
dplyr::count(dySpher.genes.presto, group)
```

****FGSEA cluster 0 (no enriched pathways)****
```{r eval=FALSE, echo=FALSE}
# use dplyr to get the appropriate gene set
msig.hs <- msigdbr(species = "Homo sapiens")

# GO biological process (GO:BP)
msig.GOBP.hs <- msig.hs %>% 
dplyr::filter(gs_subcat == "GO:BP")

head(msig.GOBP.hs)
```

```{r eval=FALSE, echo=FALSE}
fgsea_sets_GOBP <- msig.GOBP.hs %>% split(x = .$gene_symbol, f = .$gs_name)
```

```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust0 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "0" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust0) # 24 genes only!
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust0, 
          "./outputs50K/DAOY_SPHER/daoySpheroid.RiboRetained.cluster0.genes.presto.gProfilerInput.csv", quote = F)
```


```{r eval=FALSE, echo=FALSE}
ranks.clus0 <- deframe(dySpher.genes.presto.clust0)
head(ranks.clus0)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
# note: scoreType=POS only applies logFC (above) > 0
fgseaRes.clus0.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus0, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - NO enriched pathways!
fgseaResTidy.clus0.GOBP <- fgseaRes.clus0.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus0.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10)
```

****FGSEA cluster 1 (no enriched pathways)****
```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust1 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "1" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust1) # 134 genes only!
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust1, 
          "./outputs50K/DAOY_SPHER/daoySpheroid.RiboRetained.cluster1.genes.presto.gProfilerInput.csv", quote = F)
```

```{r eval=FALSE, echo=FALSE}
ranks.clus1 <- deframe(dySpher.genes.presto.clust1)
head(ranks.clus1)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
fgseaRes.clus1.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus1, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - NO significant pathways for cluster 1 - look at the high 'padj' values (>0.05)
fgseaResTidy.clus1.GOBP <- fgseaRes.clus1.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus1.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10)
```


****FGSEA cluster 2****
```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust2 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "2" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust2) # 83 genes only
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust2, 
          "./outputs50K/DAOY_SPHER/daoySpheroid.RiboRetained.cluster2.genes.presto.gProfilerInput.csv", quote = F)
```

```{r eval=FALSE, echo=FALSE}
ranks.clus2 <- deframe(dySpher.genes.presto.clust2)
head(ranks.clus2)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
fgseaRes.clus2.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus2, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - protein synthesis pathways upregulated
fgseaResTidy.clus2.GOBP <- fgseaRes.clus2.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus2.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10) 
```

```{r eval=FALSE, echo=FALSE}
# use fwrite() as a column of the tibble is a list!
fwrite(fgseaResTidy.clus2.GOBP, 
       file = "./outputs50K/DAOY_SPHER/cluster2.DAOYspheroid.multifgsea.presto.GOBiolProcessGenes.csv")
```


****FGSEA cluster 3 (no enriched pathways)****
```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust3 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "3" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust3) # 138 genes only
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust3, 
          "./outputs50K/DAOY_SPHER/daoySpheroid.RiboRetained.cluster3.genes.presto.gProfilerInput.csv", quote = F)
```

```{r eval=FALSE, echo=FALSE}
ranks.clus3 <- deframe(dySpher.genes.presto.clust3)
head(ranks.clus3)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
fgseaRes.clus3.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus3, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - no significant pathway enrichment
fgseaResTidy.clus3.GOBP <- fgseaRes.clus3.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus3.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10) # no significant pathway enrichment! (using padj < 0.05)
```


****FGSEA cluster 4****
enriched for cell cycle pathways
```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust4 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "4" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust4) # 296 genes
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust4, 
          "./outputs50K/DAOY_SPHER/daoySpheroid.RiboRetained.cluster4.genes.presto.gProfilerInput.csv", quote = F)
```

```{r eval=FALSE, echo=FALSE}
ranks.clus4 <- deframe(dySpher.genes.presto.clust4)
head(ranks.clus4)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
fgseaRes.clus4.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus4, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - enrichment for cell division/proliferation
fgseaResTidy.clus4.GOBP <- fgseaRes.clus4.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus4.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10) 
```

```{r eval=FALSE, echo=FALSE}
# use fwrite() as a column of the tibble is a list!
fwrite(fgseaResTidy.clus4.GOBP, 
       file = "./outputs50K/DAOY_SPHER/cluster4.DAOYspheroid.multifgsea.presto.GOBiolProcessGenes.csv")
```

SECTION END
##########################################################################################################

##########################################################################################################
##########################################################################################################
***Alternate workflow with removal of cells with > 10% mitochondrial gene expression, RETENTION of mitochondrial genes, and REGRESSION of mitochondrial genes***
```{r eval=FALSE, echo=FALSE}
# identify mitochondrial genes
mito.genes <- grep(pattern = "^MT", x = rownames(x = dySpher), value = TRUE)
percent.mito <- 
  Matrix::colSums(dySpher[mito.genes, ])/Matrix::colSums(dySpher)

# add the mitochondrial gene percent to the metadata
dySpher <- 
  AddMetaData(object = dySpher, metadata = percent.mito, col.name = "percent.mito")

# remove all cells with percent.mito > 10%
dySpher <- subset(dySpher, subset = percent.mito < 0.1)
dim(dySpher) # reduced to 16,238 genes and 649 cells

```

```{r eval=FALSE, echo=FALSE}
# save the NON-normalised, NON-clustered seurat object
saveRDS(dySpher, file = "./outputs50K/DAOY_SPHER/MITO_REGRESS/filt.CellsGenes.MitoGenesPresent.RiboRetained.seurat.DAOYspher.rds")
```

```{r eval=FALSE, echo=FALSE}
# load cell cycle genes, calculate cell-cycle difference in advance of regressing out CC difference

# normalise object before running cell cycle scores as per Seurat github issue: https://github.com/satijalab/seurat/issues/1679
dySpher <- NormalizeData(dySpher)

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
# calculate CC difference
dySpher <- CellCycleScoring(dySpher,
                g2m.features = g2m.genes,
                s.features = s.genes)
dySpher$CC.difference <- dySpher$S.Score - dySpher$G2M.Score
```

```{r eval=FALSE, echo=FALSE}
# this object (dySpher.input.integration) is input for integration - already SCTransformed, cell cycle difference regressed, and RunPCA()
dySpher.input.integration <- SCTransform(dySpher, 
                   vars.to.regress = c("CC.difference", "percent.mito"),
                   method = "glmGamPoi",
                   vst.flavor = "v2",
                   verbose = FALSE) %>% 
  RunPCA(npcs = 30, verbose = FALSE)
```

```{r eval=FALSE, echo=FALSE}
# saved the UNCLUSTERED SCTransformed object, CC difference regressed, PCA reduction, as input for integrated workflow
saveRDS(dySpher.input.integration, "./outputs50K/DAOY_SPHER/MITO_REGRESS/DAOYspher.inputIntegration.RiboRetained.SCTransformed.MitoRegressed.CCdiffRegressed.PCA.unclustered.rds")
```


```{r eval=FALSE, echo=FALSE}
# cluster the cells with sctransform workflow
dySpher <- SCTransform(dySpher, 
                       vars.to.regress = c("CC.difference", "percent.mito"),
                       method = "glmGamPoi", verbose = FALSE) %>% 
  RunPCA(verbose = FALSE) %>% 
  RunUMAP(dims=1:30, verbose=FALSE) %>% 
  FindNeighbors(dims=1:30, verbose=FALSE) %>% 
  FindClusters(verbose=FALSE, resolution = 0.5)
```

```{r eval=FALSE, echo=FALSE, DimplotDAOYspherUmapSCTransformMitoRegress, fig.height=6, fig.width=10}
DimPlot(dySpher, label = TRUE, pt.size = 1, label.size = 4)
```

```{r eval=FALSE, echo=FALSE}
# save the clustered, SCTransformed object above
saveRDS(dySpher, file = "./outputs50K/DAOY_SPHER/MITO_REGRESS/DAOYspher.MitoCellsGenesFilt.RiboRetained.SCT.clustered.MitoRegressed.CCdiffRegressed.rds")
```

```{r eval=FALSE, echo=FALSE}
# load the list of proliferation genes in Seurat
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
prolif.genes <- list(c(s.genes, g2m.genes))
```


```{r eval=FALSE, echo=FALSE, AddModuleScoreProlifernDAOYspheroidMitoRegress, fig.height=6, fig.width=10}
# Dimplot of UMAP clusters with proliferation score superimposed (as heatmap)
phase.dySpher <- AddModuleScore(dySpher,
                        features = prolif.genes,
                        name = "prolif.genes")
FeaturePlot(phase.dySpher, features = 'prolif.genes1', label = TRUE, repel = TRUE, pt.size = 1,
            label.size = 4) +
   scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "")
```

****Get the cluster markers****
```{r eval=FALSE, echo=FALSE}
# get the markers of each cluster
dySpher.POSmarkers <- FindAllMarkers(object = dySpher, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r eval=FALSE, echo=FALSE}
# extract only the top 30 markers of each cluster
# cluster 0 only is enriched for ribosome genes.
dySpher.POSmarkers %>% 
  dplyr::group_by(cluster) %>% 
  slice_head(n = 30)
```

```{r eval=FALSE, echo=FALSE}
# Check expression of granule markers: MEIS1 is expressed of these seven genes, filtering out the mitochondrial genes in the original workflow made no difference to this result, unsurprisngly.
dySpher.POSmarkers %>%
  filter(gene %in% c("NEUROD1", "CNTN1", "CNTN2", "MEIS1", "MEIS2", "DCX", "SATB2"))
```

```{r eval=FALSE, echo=FALSE, ViolinplotDAOYspheroidMKI67andMEIS1}
# rough-and-ready plot
VlnPlot(dySpher, features = c("MKI67", "MEIS1"))
```

****FGSEA to find gene ontology pathway enrichment for the clusters****
```{r eval=FALSE, echo=FALSE}
# prepare the data for gProfiler (web app) input - gProfiler output then used as Cytoscape input
dySpher.genes.presto <- wilcoxauc(dySpher, 'seurat_clusters')
nrow(dySpher.genes.presto)
dplyr::count(dySpher.genes.presto, group)
```

*****FGSEA cluster 0*****
Enriched for protein translation/synthesis
```{r eval=FALSE, echo=FALSE}
# use dplyr to get the appropriate gene set
msig.hs <- msigdbr(species = "Homo sapiens")

# GO biological process (GO:BP)
msig.GOBP.hs <- msig.hs %>% 
dplyr::filter(gs_subcat == "GO:BP")

head(msig.GOBP.hs)
```

```{r eval=FALSE, echo=FALSE}
fgsea_sets_GOBP <- msig.GOBP.hs %>% split(x = .$gene_symbol, f = .$gs_name)
```

```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust0 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "0" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust0) # 76 genes only!
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust0, 
          "./outputs50K/DAOY_SPHER/MITO_REGRESS/daoySpheroid.RiboRetained.MitoRegress.cluster0.genes.presto.gProfilerInput.csv", quote = F)
```


```{r eval=FALSE, echo=FALSE}
ranks.clus0 <- deframe(dySpher.genes.presto.clust0)
head(ranks.clus0)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
# note: scoreType=POS only applies logFC (above) > 0
fgseaRes.clus0.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus0, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - NO enriched pathways!
fgseaResTidy.clus0.GOBP <- fgseaRes.clus0.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus0.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10)
```

```{r eval=FALSE, echo=FALSE}
# use fwrite() as a column of the tibble is a list!
fwrite(fgseaResTidy.clus0.GOBP, 
       file = "./outputs50K/DAOY_SPHER/MITO_REGRESS/cluster0.DAOYspheroid.multifgsea.presto.GOBiolProcessGenes.csv")
```


*****FGSEA cluster 1*****
(no enriched pathways)
```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust1 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "1" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust1) # 52 genes only!
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust1, 
          "./outputs50K/DAOY_SPHER/MITO_REGRESS/daoySpheroid.RiboRetained.cluster1.genes.presto.gProfilerInput.csv", quote = F)
```

```{r eval=FALSE, echo=FALSE}
ranks.clus1 <- deframe(dySpher.genes.presto.clust1)
head(ranks.clus1)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
fgseaRes.clus1.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus1, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - NO significant pathways for cluster 1 - look at the high 'padj' values (>0.05)
fgseaResTidy.clus1.GOBP <- fgseaRes.clus1.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus1.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10)
```

```{r eval=FALSE, echo=FALSE}
# use fwrite() as a column of the tibble is a list!
fwrite(fgseaResTidy.clus1.GOBP, 
       file = "./outputs50K/DAOY_SPHER/MITO_REGRESS/cluster1.DAOYspheroid.multifgsea.presto.GOBiolProcessGenes.csv")
```

*****FGSEA cluster 2*****
(no enriched pathways)
```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust2 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "2" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust2) # 136 genes
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust2, 
          "./outputs50K/DAOY_SPHER/MITO_REGRESS/daoySpheroid.RiboRetained.cluster2.genes.presto.gProfilerInput.csv", quote = F)
```

```{r eval=FALSE, echo=FALSE}
ranks.clus2 <- deframe(dySpher.genes.presto.clust2)
head(ranks.clus2)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
fgseaRes.clus2.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus2, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - protein synthesis pathways upregulated
fgseaResTidy.clus2.GOBP <- fgseaRes.clus2.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus2.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10) 
```

```{r eval=FALSE, echo=FALSE}
# use fwrite() as a column of the tibble is a list!
fwrite(fgseaResTidy.clus2.GOBP, 
       file = "./outputs50K/DAOY_SPHER/MITO_REGRESS/cluster2.DAOYspheroid.multifgsea.presto.GOBiolProcessGenes.csv")
```

*****FGSEA cluster 3*****
enriched for cell cycle pathways
```{r eval=FALSE, echo=FALSE}
# filter and order the gene list
dySpher.genes.presto.clust3 <- dySpher.genes.presto %>% 
  dplyr::filter(group == "3" & logFC > 0.25) %>%
  dplyr::arrange(desc(auc)) %>% 
  dplyr::select(feature, auc)

nrow(dySpher.genes.presto.clust3) # 267 genes 
```

```{r eval=FALSE, echo=FALSE}
# gProfiler input - use Filezilla to pull the file to my Mac and then copy and paste the ordered list
# of genes into gProfiler web app
write.csv(dySpher.genes.presto.clust3, 
          "./outputs50K/DAOY_SPHER/MITO_REGRESS/daoySpheroid.RiboRetained.cluster3.genes.presto.gProfilerInput.csv", quote = F)
```

```{r eval=FALSE, echo=FALSE}
ranks.clus3 <- deframe(dySpher.genes.presto.clust3)
head(ranks.clus3)
```

```{r eval=FALSE, echo=FALSE}
# use fgsea 'multilevel' function; the multilevel function has by default NPermSimple=1000
fgseaRes.clus3.GOBP <- 
  fgseaMultilevel(fgsea_sets_GOBP, stats = ranks.clus3, eps = 0, minSize = 15, scoreType = "pos", maxSize = 200)
```

```{r eval=FALSE, echo=FALSE}
# tidy up the data - no significant pathway enrichment
fgseaResTidy.clus3.GOBP <- fgseaRes.clus3.GOBP %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy.clus3.GOBP %>% 
  dplyr::select(-leadingEdge, -ES) %>% 
  arrange(padj) %>% 
  head(10) # cell cycle pathway enrichment (using padj < 0.05)
```

SECTION END
##########################################################################################################

##########################################################################################################
##########################################################################################################
***Alternate workflow with removal of cells with > 10% mitochondrial gene expression, RETENTION of mitochondrial genes (WITHOUT regression)***
the result is the same as with regression of mitochondrial genes, the same number of clusters are found. There are no clusters whose top markers are mitochondrial genes. In view of the similarity to the results of mitochondrial gene regression, no functional enrichment (using FGSEA) was performed.
```{r eval=FALSE, echo=FALSE}
# identify mitochondrial genes
mito.genes <- grep(pattern = "^MT", x = rownames(x = dySpher), value = TRUE)
percent.mito <- 
  Matrix::colSums(dySpher[mito.genes, ])/Matrix::colSums(dySpher)

# add the mitochondrial gene percent to the metadata
dySpher <- 
  AddMetaData(object = dySpher, metadata = percent.mito, col.name = "percent.mito")

# remove all cells with percent.mito > 10%
dySpher <- subset(dySpher, subset = percent.mito < 0.1)
dim(dySpher) # reduced to 16,238 genes and 649 cells
```

```{r eval=FALSE, echo=FALSE}
# save the NON-normalised, NON-clustered seurat object
saveRDS(dySpher, file = "./outputs50K/DAOY_SPHER/MITO_RETAIN/filt.CellsGenes.MitoGenesUNregressed.RiboRetained.seurat.DAOYspher.rds")
```

```{r eval=FALSE, echo=FALSE}
# load cell cycle genes, calculate cell-cycle difference in advance of regressing out CC difference

# normalise object before running cell cycle scores as per Seurat github issue: https://github.com/satijalab/seurat/issues/1679
dySpher <- NormalizeData(dySpher)

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
# calculate CC difference
dySpher <- CellCycleScoring(dySpher,
                g2m.features = g2m.genes,
                s.features = s.genes)
dySpher$CC.difference <- dySpher$S.Score - dySpher$G2M.Score
```

```{r eval=FALSE, echo=FALSE}
# this object (dySpher.input.integration) is input for integration - already SCTransformed, cell cycle difference regressed, and RunPCA()
dySpher.input.integration <- SCTransform(dySpher, 
                   vars.to.regress = c("CC.difference"),
                   method = "glmGamPoi",
                   vst.flavor = "v2",
                   verbose = FALSE) %>% 
  RunPCA(npcs = 30, verbose = FALSE)
```

```{r eval=FALSE, echo=FALSE}
# saved the UNCLUSTERED SCTransformed object, CC difference regressed, PCA reduction, as input for integrated workflow
saveRDS(dySpher.input.integration, "./outputs50K/DAOY_SPHER/MITO_RETAIN/DAOYspher.inputIntegration.RiboRetained.SCTransformed.MitoUNregressed.CCdiffRegressed.PCA.unclustered.rds")
```

```{r eval=FALSE, echo=FALSE}
# cluster the cells with sctransform workflow
dySpher <- SCTransform(dySpher, 
                       vars.to.regress = c("CC.difference"),
                       method = "glmGamPoi", verbose = FALSE) %>% 
  RunPCA(verbose = FALSE) %>% 
  RunUMAP(dims=1:30, verbose=FALSE) %>% 
  FindNeighbors(dims=1:30, verbose=FALSE) %>% 
  FindClusters(verbose=FALSE, resolution = 0.5)
```

```{r eval=FALSE, echo=FALSE, DimplotDAOYspherUmapSCTransformMitoRegress, fig.height=6, fig.width=10}
DimPlot(dySpher, label = TRUE, pt.size = 1, label.size = 4)
```

****Get the cluster markers****
```{r eval=FALSE, echo=FALSE}
# get the markers of each cluster
dySpher.POSmarkers <- FindAllMarkers(object = dySpher, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r eval=FALSE, echo=FALSE}
# extract only the top 30 markers of each cluster
# cluster 0 only is enriched for ribosome genes.
dySpher.POSmarkers %>% 
  dplyr::group_by(cluster) %>% 
  slice_head(n = 30)
```

```{r eval=FALSE, echo=FALSE}
# Check expression of granule markers: MEIS1 is expressed of these seven genes, filtering out the mitochondrial genes in the original workflow made no difference to this result, unsurprisngly.
dySpher.POSmarkers %>%
  filter(gene %in% c("NEUROD1", "CNTN1", "CNTN2", "MEIS1", "MEIS2", "DCX", "SATB2"))
```

```{r eval=FALSE, echo=FALSE}
# save the clustered, SCTransformed object above
saveRDS(dySpher, file = "./outputs50K/DAOY_SPHER/MITO_RETAIN/DAOYspher.MitoCellsGenesFilt.RiboRetained.SCT.clustered.MitoUNregressed.CCdiffRegressed.rds")
```

```{r eval=FALSE, echo=FALSE}
# load the list of proliferation genes in Seurat
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
prolif.genes <- list(c(s.genes, g2m.genes))
```

```{r eval=FALSE, echo=FALSE, AddModuleScoreProlifernDAOYspheroidMitoRegress, fig.height=6, fig.width=10}
# Dimplot of UMAP clusters with proliferation score superimposed (as heatmap)
phase.dySpher <- AddModuleScore(dySpher,
                        features = prolif.genes,
                        name = "prolif.genes")
FeaturePlot(phase.dySpher, features = 'prolif.genes1', label = TRUE, repel = TRUE, pt.size = 1,
            label.size = 4) +
   scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "")
```

```{r eval=FALSE, echo=FALSE}
sessionInfo()
```


SECTION END
##########################################################################################################

